src/utils/RNPxlSearch.cpp

```diff
@@ -1391,7 +1391,15 @@ class RNPxlSearch :
 
           if (!purities.empty())
           {
-            ph.setMetaValue("precursor_purity", purities[scan_index].signal_proportion);
+            map<String, PrecursorPurity::PurityScores>::const_iterator it = purities.find(exp[scan_index].getNativeID());
+            if (it != purities.end())
+            {
+              ph.setMetaValue("precursor_purity", it->second.signal_proportion);
+            }
+            else // probably not necessary
+            {
+              ph.setMetaValue("precursor_purity", 0);
```

Comment:
would assuming a perfect purity (1.0?) be better?

Reply:
I don't think so. This is the case where we don't have any information about the precursor. 1 would  imply prefect purity and happens quite often with real data, while 0 would be extremely rare and a weird glitch in a case where we have data. I would just write the code using these values in a way that ignores values of 0.